resource/cmake/DdsFunctions.cmake

```
@@ -13,29 +13,29 @@
 #                                                                             #
 ###############################################################################
 
-function(connextdds_run_codegen)
-    set(BOOLEANS)
-    set(SINGLE_VALUE_ARGS IDL_FILE LANGUAGE OUTPUT_DIR)
-    set(MULTI_VALUE_ARGS INPUT_PARAMS)
+function(rtigw_connextdds_run_codegen)
+    set(_BOOLEANS)
+    set(_SINGLE_VALUE_ARGS IDL_FILE LANGUAGE OUTPUT_DIR)
+    set(_MULTI_VALUE_ARGS INPUT_PARAMS)
```

Consider renaming this function to something more "function oriented" (also `codegen` is not thing, users know `rtiddsgen`, or maybe `ddsgen`). Also simplify arguments similarly to what we did for other functions (and some name adjustments, most importantly `INPUT_PARAMS`)

```suggestion
function(rtigw_generate_type_support)
    set(_BOOLEANS)
    set(_SINGLE_VALUE_ARGS LANGUAGE INPUT OUTPUT)
    set(_MULTI_VALUE_ARGS OPTIONS)
```

---

I've done some changes following the same structure that the complementary file in the common-cmake repo: https://github.com/rticommunity/rticonnextdds-cmake-utils/blob/main/cmake/Modules/ConnextDdsCodegen.cmake

---

I agree that we should keep this function in line with the one offered by `rticonnextdds-cmake-utils`, and I was about to say "maybe we can just copy the implementation from there?", but actually: why don't we just add the repo as a submodule and use the functions it provides already?

This PR could be an opportunity to do that, and start with replacing the `run_codegen()` function from this repo with `connextdds_rtiddsgen_run()`. We can also rely on the updated `FindRTIConnextDDS.cmake`. What do you think?

---

We can add rticommunity/rticonnextdds-cmake-utils as another submodule under `third-party/`, or we could introduce another directory for RTI modules (e.g. `common/modules/`), but I think I'd rather go with the first option.

---

I thought about doing it in a different PR, but adding it here makes sense. About the directory, I thought adding it in `resource/cmake/rticonnextdds-cmake-utils` and leave the custom cmake files where they currently are: `resource/cmake`

---

added `rticonnextdds-cmake-utils` under `third-party`